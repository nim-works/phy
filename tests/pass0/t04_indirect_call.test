discard """
  output: "(Done 100)"
"""
(TypeDefs
  (ProcTy (Int 8))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 0) 0 (Locals)
    (List
      (Block (Params)
        (Return (IntVal 100))
      )
    )
  )
  (ProcDef (Type 0) 0
    (Locals (Int 8))
    (List
      (Block (Params)
        (Asgn (Local 0) (Proc 0))
        (Return (Call (Type 0) (Copy (Local 0))))
      )
    )
  )
)
