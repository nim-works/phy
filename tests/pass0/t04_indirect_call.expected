.type t0 () -> int
.start t0 p0
  LdImmInt 100
  Ret
.end
.start t0 p1
  LdImmInt 1
  IndCall t0 0
  Ret
.end
